app/src/main/java/com/android/sample/model/workout/WorkoutViewModel.kt

```diff
@@ -27,7 +52,17 @@ open class WorkoutViewModel<out T : Workout>(private val repository: WorkoutRepo
 
   /** Gets all Workout documents. */
   fun getWorkouts() {
-    repository.getDocuments(onSuccess = { workouts_.value = it }, onFailure = {})
+    viewModelScope.launch {
+      repository.getDocuments(
+          onSuccess = { fetchedWorkouts ->
+            // Filter the fetched workouts by the workoutClass
+            val filteredWorkouts = fetchedWorkouts.filter { workoutClass.isInstance(it) }.map { it }
```

Comment:
Does the map do anything ?

Reply:
Nice catch again i forgot to remove this one too